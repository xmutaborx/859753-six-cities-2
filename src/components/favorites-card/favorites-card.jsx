import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import Operations from '../../store/operations';
import {correctRating} from '../../helpers/helpers';

const FavoritesCard = (props) => {
  const {title, price, type, image, rating, id, isFavorite, onToggleFavorites} = props;

  return (
    <article className="favorites__card place-card">
      <div className="favorites__image-wrapper place-card__image-wrapper">
        <Link to={`/offer/${id}`}>
          <img className="place-card__image" src={image} width={150} height={110} alt="Place image" />
        </Link>
      </div>
      <div className="favorites__card-info place-card__info">
        <div className="place-card__price-wrapper">
          <div className="place-card__price">
            <b className="place-card__price-value">â‚¬{price}</b>
            <span className="place-card__price-text">/&nbsp;night</span>
          </div>
          <button
            className={`place-card__bookmark-button button ${isFavorite ? `place-card__bookmark-button--active` : ``}`}
            type="button"
            onClick={() => onToggleFavorites(id, !isFavorite)}
          >
            <svg className="place-card__bookmark-icon" width={18} height={19}>
              <use xlinkHref="#icon-bookmark" />
            </svg>
            <span className="visually-hidden">In bookmarks</span>
          </button>
        </div>
        <div className="place-card__rating rating">
          <div className="place-card__stars rating__stars">
            <span style={{width: `${correctRating(rating)}%`}} />
            <span className="visually-hidden">Rating</span>
          </div>
        </div>
        <h2 className="place-card__name">
          <Link to={`/offer/${id}`}>{title}</Link>
        </h2>
        <p className="place-card__type">{type}</p>
      </div>
    </article>
  );
};

FavoritesCard.propTypes = {
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  id: PropTypes.number.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  onToggleFavorites: PropTypes.func,
};

const mapDispatchToProps = (dispatch) => ({
  onToggleFavorites: (id, status) => dispatch(Operations.toggleFavorites(id, status)),
});

export {FavoritesCard};
export default connect(null, mapDispatchToProps)(FavoritesCard);
