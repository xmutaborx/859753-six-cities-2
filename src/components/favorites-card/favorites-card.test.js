import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import {FavoritesCard} from './favorites-card.jsx';

it(`Favorites Card render correctly`, () => {
  const tree = renderer
    .create(<Router>
      <FavoritesCard
        title={`title`}
        price={69}
        type={`room`}
        image={`src`}
        rating={5}
        id={1}
        isFavorite={true}
        onToggleFavorites={() => {}}
      />
    </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
