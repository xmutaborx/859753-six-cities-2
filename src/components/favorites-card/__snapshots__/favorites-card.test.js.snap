// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Card render correctly 1`] = `
<article
  className="favorites__card place-card"
>
  <div
    className="favorites__image-wrapper place-card__image-wrapper"
  >
    <a
      href="/offer/1"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={110}
        src="src"
        width={150}
      />
    </a>
  </div>
  <div
    className="favorites__card-info place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          69
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button button place-card__bookmark-button--active"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height={19}
          width={18}
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        title
      </a>
    </h2>
    <p
      className="place-card__type"
    >
      room
    </p>
  </div>
</article>
`;
